docs/adr/adr-013-tail-padding-optimizations.md

```diff
@@ -0,0 +1,55 @@
+# ADR 13: Tail Padding Optimizations
+
+## Changelog
+
+- 24.02.2023: First Draft
+
+## Author
+
+@nashqueue
+
+## Context
+
+If the Celestia block is not full it will create tail padding. Tail padded shares look always the same. Tail padded shares create tail padded rows. tail padded rows erasure to erasure-coded tail padded rows. Tail padded rows combined with erasure-coded tail padded rows result in a full tail padded row. A full tail padded row of the same length will always have the same Namespace Merkle Trie on top of it. This will result in a tail padded row root.
+All of those things are deterministic and can be computed beforehand.
+
+Given a tail padded row root inside the DA-Header you can deduce that the underlying row is a full tail padded row. This information can be beneficial in many ways.
+
+### DAS
+
+If a Light Node receives a tail padded row root in the DAH it does not have to DAS this root. It knows the underlying shares and could generate the shares and path itself.
+
+Hypothesis: The more tail padded rows exist the fewer samples a Light Node has to do for the same DA guarantee.
+
+### Block reconstruction
+
+Given a tail padded rowroot you can use the predetermined shares to fill up the store. You will not need to request those shares from light nodes. This will save bandwidth and reconstruction time.
+
+### Storage
+
+You don`t need to save copies of the same shares multiple times. Given that 3/4 of the ODS can be tail padding it can save up to 3/16 + 3/16 = 6/16 = 3/8 of storage of the EDS.
```

Comment:
```suggestion
You don't need to save copies of the same shares multiple times. Given that 3/4 of the ODS can be tail padding it can save up to 3/16 + 3/16 = 6/16 = 3/8 of storage of the EDS.
```

Reply:
To clarify: this is `3/16` instead of `12/16` of the EDS because the shares in quadrants 1, 2, 3 won't have any share that is exactly a tail padding share (due to the process of erasure coding) - right?

Reply:
So in quadrant one you have the ODS . In ODS you can have max 3/4 tail padding which ich 3/16 of all quadrants. Erasurecoding the same thing gets you the same. So you will have 3/4 of Quadrant 2 (the one right to Q1) be the same. That's how you get the second 3/16

Reply:
> Erasurecoding the same thing gets you the same.

Wow that's a very interesting property that I did not expect. Thanks for sharing. I verified by debugging a process proposal test. For a square size of 4 where the bottom two rows of the square are tail padding, in the extended data square the bottom two rows of Q2 are also tail padding.

## Screenshot
<img width="930" alt="Screenshot 2023-02-24 at 7 46 46 PM" src="https://user-images.githubusercontent.com/3699047/221326298-09c2534b-f43c-4edb-b99b-6f24191c444b.png">

Reply:
Wait, maybe I misunderstand my explanation. Can you explain the screenshot? What do you mean by also tail padding?

Reply:
**Legend**
B = PFB tx compact share
R = Reserved namespace padding share
1 = Blob 1 share
2 = Blob 2 share
3 = Blob 3 share
T = Tail padding share
P = Parity share


In the test we have an original data square of:
```ascii
BBR1
2233
TTTT
TTTT
```

and the extended data square's Q0 and Q1 looks like

```ascii
BBR1 | PPPP
2233 | PPPP
TTTT | TTTT
TTTT | TTTT
```

TBH I'm surprised by the observation that the Q1 shares in the top two rows look like parity shares (i.e. they aren't identical to the corresponding share from Q0) but the shares in the bottom two rows of Q1 look identical to tail padding shares.

Reply:
wow nice observation. That is actually really interesting. I was expecting the erasure-coded tailpadding different from the tailpadding. Every share the same but different from T. But this looks so intuitive rn :D Good catch.